### PR TITLE
attempt to fix type alias inequality #5360 and #5238

### PR DESCRIPTION
**Pull request by jangko**

### CODE REVIEW

**Comment by Araq** [comment]
FAIL: tenum.nim
Test "tests/enum/tenum.nim" in category "enum"
Failure: reNimcCrash
Expected:
Gotten:
Hint: used config file '/home/travis/build/nim-lang/Nim/config/nim.cfg' [Conf]
tenum.nim(13, 29) template/generic instantiation from here
lib/core/macros.nim(269, 22) Error: redefinition of 'geItem1'

**Comment by jangko** [comment]
yup, I'am still figuring it out

File: compiler/semtypes.nim [hunk 1]
```diff
@@ -1138,14 +1138,22 @@ proc semGeneric(c: PContext, n: PNode, s: PSym, prev: PType): PType =
...
   if n.typ.kind == tyTypeDesc:
     result = n.typ.base
-    # fix types constructed by macros:
-    if prev != nil and prev.sym != nil and result.sym.isNil:
-      result.sym = prev.sym
-      result.sym.typ = result
+    # fix types constructed by macros/template:
+    if prev != nil and prev.sym != nil:
+      if result.sym.isNil:
+        # types constructed by macros
```
Comment (Araq) [inline compiler/semtypes.nim]: ``result.sym.isNil`` is hardly an indicator for "type constructed by macro". This comment is confusing, improve it please.

**Comment by jangko** [comment]
sorry for my terribly broken English. I learned most of my vocabulary from free online English novels. not from proper education at school.

**Comment by Araq** [comment]
Not what I meant, but good enough.

**Pull request merged**